### ISSUE TITLE
ProviderError: invalid opcode: INVALID

### ISSUE THREAD

**Issue by TeamRainless** [PatrickAlphaC/all-on-chain-generated-nft#30]
So I'm at the part where the contract has been issued and upload and a verification link is given and it's just about to mint the NFT and I get this error with no clear error message:

An unexpected error occurred:

```
Error: ERROR processing /Users/rainless/Downloads/all-on-chain-generated-nft/deploy/02_Deploy_RandomSVG.js:
ProviderError: invalid opcode: INVALID
    at HttpProvider.request (/Users/rainless/Downloads/all-on-chain-generated-nft/node_modules/hardhat/src/internal/core/providers/http.ts:74:19)
```

This is what's on that line in http.ts:

 `const error = new ProviderError("HttpProviderError", -1);`

...but I'm not sure how that helps us.  And I have no idea how to get around this error message.

**Comment by PatrickAlphaC** [comment]
Thanks for making the issue.

Could you please send:

1. What command did you run?
2. Have you edited the codebase?
3. What network are you working on?

**Comment by TeamRainless** [comment]
1. hh deploy --network rinkeby
2. I changed nothing (besides my own environment variables of course.)
3. Rinkeby

**Comment by TeamRainless** [comment]
Here's the complete error message:

```
reusing "RandomSVG" at 0xa20C316721F71546FE12afE8292a22F5E3000f35
You have deployed an NFT contract to 0xa20C316721F71546FE12afE8292a22F5E3000f35
Verify with:
 npx hardhat verify --network rinkeby 0xa20C316721F71546FE12afE8292a22F5E3000f35 0xb3dCcb4Cf7a26f6cf6B120Cf5A73875B7BBc655B 0x01be23585060835e02b77ef475b0cc51aa1e0709 0x2ed0feb3e7fd2022120aa84fab1945545a9f2ffc9076fd6156fa96eaff4c1311 100000000000000000
Duplicate definition of Transfer (Transfer(address,address,uint256,bytes), Transfer(address,address,uint256))
An unexpected error occurred:

Error: ERROR processing /Users/rainless/Downloads/all-on-chain-generated-nft/deploy/02_Deploy_RandomSVG.js:
ProviderError: invalid opcode: INVALID
    at HttpProvider.request (/Users/rainless/Downloads/all-on-chain-generated-nft/node_modules/hardhat/src/internal/core/providers/http.ts:74:19)
    at LocalAccountsProvider.request (/Users/rainless/Downloads/all-on-chain-generated-nft/node_modules/hardhat/src/internal/core/providers/accounts.ts:188:34)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at EthersProviderWrapper.send (/Users/rainless/Downloads/all-on-chain-generated-nft/node_modules/@nomiclabs/hardhat-ethers/src/internal/ethers-provider-wrapper.ts:13:20)
    at DeploymentsManager.executeDeployScripts (/Users/rainless/Downloads/all-on-chain-generated-nft/node_modules/hardhat-deploy/src/DeploymentsManager.ts:1014:19)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at DeploymentsManager.runDeploy (/Users/rainless/Downloads/all-on-chain-generated-nft/node_modules/hardhat-deploy/src/DeploymentsManager.ts:849:5)
    at Environment._runTaskDefinition (/Users/rainless/Downloads/all-on-chain-generated-nft/node_modules/hardhat/src/internal/core/runtime-environment.ts:219:14)
    at Environment.run (/Users/rainless/Downloads/all-on-chain-generated-nft/node_modules/hardhat/src/internal/core/runtime-environment.ts:131:14)
    at SimpleTaskDefinition.action (/Users/rainless/Downloads/all-on-chain-generated-nft/node_modules/hardhat-deploy/src/index.ts:446:32)
    at Environment._runTaskDefinition (/Users/rainless/Downloads/all-on-chain-generated-nft/node_modules/hardhat/src/internal/core/runtime-environment.ts:219:14)
    at Environment.run (/Users/rainless/Downloads/all-on-chain-generated-nft/node_modules/hardhat/src/internal/core/runtime-environment.ts:131:14)
    at SimpleTaskDefinition.action (/Users/rainless/Downloads/all-on-chain-generated-nft/node_modules/hardhat-deploy/src/index.ts:521:5)
    at Environment._runTaskDefinition (/Users/rainless/Downloads/all-on-chain-generated-nft/node_modules/hardhat/src/internal/core/runtime-environment.ts:219:14)

```

**Comment by TeamRainless** [comment]
I'm curious why "invalid opcode" doesn't give me a more verbose error response than: "INVALID." Like WHAT, exactly, is invalid? Where am I supposed to look to track down the problem?

**Comment by TeamRainless** [comment]
Any update or idea?

**Comment by PatrickAlphaC** [comment]
Yeah javascript has a hard time giving errors often :( 

Usually it means you've got some wrong solidty or JS. I can't seem to replicate your issue. I just recloned and ran `hh deploy --network rinkeby` myself and everything went through.

Maybe you ran out of gas?

**Comment by TeamRainless** [comment]
Well maybe.  How could I fix that? I don't understand how it could deploy the contract just fine but not deploy the NFT. Could it have been related to the fact that I didn't have any LINK at the time?  I went back and gave myself some LINK (had to add the currency and run the faucet and all that...) but now it doesn't do anything at all. I can't deploy the contract (before and after recloning and starting from scratch) OR the NFT now. It works deploying on a local network with hh deploy but hh deploy --network rinkeby just hangs.

**Comment by PatrickAlphaC** [comment]
You could have run out of ETH in your wallet that is deploying. Get more rinkeby ETH from the faucet. \

More info: https://www.youtube.com/watch?v=Yh8cHUB-KoU

**Comment by TeamRainless** [comment]
1. The "hh deploy" code from the beginning of the tutorial (exactly as
typed).
2. I changed nothing.
3. Rinkeby

I followed the tutorial... deployed the simple and random SVG files
locally. Then made and verified the contract for deploying externally.
Right after the part where the contract is issued, then verified, but
before the nft is minted... it gives that error. The contract shows up on
OpenSea... but not the NFT.
